### PR TITLE
D9-11: Create custom entity type.

### LINKED ISSUE #15: #11 ­- Creating a Custom D8 Content Entity Type
## Objective
At the end of this course, you will be able to

create a content entity type with administration management pages
## Exercise
Create a custom content entity type called “Contact” with the following fields:
- Name
- Email Address
- Telephone
- Address
Build / Provide a minimal CRUD management page to manage the custom contact entities
Explore why would you even want to define a custom entity type vs using a Content Type

### PR DESCRIPTION
**Pull request by amansrivastava**
#15

### CODE REVIEW

File: docroot/modules/custom/custom_contact/src/Entity/Contact.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ *     "add-form" = "/admin/content/contact/add",
+ *     "canonical" = "/contact/{contact}",
+ *     "edit-form" = "/admin/content/contact/{contact}/edit",
+ *     "delete-form" = "/admin/content/contact/{contact}/delete",
+ *     "collection" = "/admin/content/contact"
+ *   },
+ *   field_ui_base_route = "entity.contact.settings"
+ * )
+ */
+class Contact extends ContentEntityBase implements ContactInterface {
```
Comment (joshirohit100) [inline docroot/modules/custom/custom_contact/src/Entity/Contact.php:47]: Is this entity revisionable and fieldable? What all are base fields of entity?

File: docroot/modules/custom/custom_contact/src/Form/ContactSettingsForm.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Drupal\custom_contact\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configuration form for a contact entity type.
+ */
+class ContactSettingsForm extends FormBase {
```
Comment (joshirohit100) [inline docroot/modules/custom/custom_contact/src/Form/ContactSettingsForm.php]: what is purpose of this?

File: docroot/modules/custom/custom_contact/src/Entity/Contact.php (hunk 1 above)
Comment (amansrivastava) [reply docroot/modules/custom/custom_contact/src/Entity/Contact.php:47]: It is fieldable and not revisionable.
Base fields are name, email, telephone and address.

File: docroot/modules/custom/custom_contact/src/Form/ContactSettingsForm.php (hunk 2 above)
Comment (amansrivastava) [reply docroot/modules/custom/custom_contact/src/Form/ContactSettingsForm.php]: It is scaffolded from drush command to generate custom entity.

**Pull request merged**